source/org/zfin/publication/repository/HibernatePublicationRepository.java

```diff
@@ -150,18 +148,20 @@ public PaginationResult<HighQualityProbe> getHighQualityProbeNames(Term term, in
 
     //TODO: refactor to JPA Criteria?
     //TODO: ScrollableResults makes it difficult to refactor to Tuple-based hql
+
     /**
      * Note: firstRow must be 1 or greater, i.e. the way a user would describes
      * the record number. Hibernate starts with the first row numbered '0'.
      * Do not include records where the gene or probe is WITHDRAWN.
      * Do not include records where the probe is Chimeric.
      * Written in native SQL because need to order by number of figures.
      *
-     * @param anatomyTerm     anatomy term
-     * @param firstRow        first row
-     * @param numberOfRecords number
+     * @param anatomyTerm anatomy term
+     * @param pagination  Pagination bean
      */
-    public PaginationResult<MarkerStatistic> getAllExpressedMarkers(GenericTerm anatomyTerm, int firstRow, int numberOfRecords) {
+    public PaginationResult<MarkerStatistic> getAllExpressedMarkers(GenericTerm anatomyTerm, Pagination pagination) {
+        int firstRow = pagination.getStart();

```

Comment:
Looks like this getStart() method has some logic that has caused a test to start failing. It ends up returning a negative value.

https://trunk.zfin.org/build/job/6.%20TRUNK%20test/3163/testReport/junit/org.zfin.publication.repository/PublicationRepositoryTest/runGetAllExpressedGenes/